assets/src/edit-story/components/panels/design/preset/colorPreset/colorPresetActions.js

```diff
@@ -70,35 +130,109 @@ function ColorPresetActions({ color }) {
-        if (findMatchingColor(toAdd, globalStoryStyles, isText)) {
+        const currentStyles = showLocalColors
+          ? currentStoryStyles
+          : globalStoryStyles;
+        if (findMatchingColor(toAdd, currentStyles, isText)) {
           return;
         }
-
+        const newProps = showLocalColors
+          ? {
+              currentStoryStyles: {
+                ...currentStoryStyles,
+                colors: [...localColors, toAdd],
+              },
+            }
+          : {
+              globalStoryStyles: {
+                ...globalStoryStyles,
+                colors: [...globalColors, toAdd],
+              },
+            };
         updateStory({
-          properties: {
-            globalStoryStyles: {
-              ...globalStoryStyles,
-              colors: [...colors, toAdd],
-            },
-          },
+          properties: newProps,
         });
       }
     },
-    [globalStoryStyles, isText, colors, updateStory]
+    [
+      currentStoryStyles,
+      globalStoryStyles,
+      localColors,
+      showLocalColors,
+      isText,
+      globalColors,
+      updateStory,
+    ]
   );
 
+  const options = [
+    {
+      id: LOCAL,
+      name: __('Current story', 'web-stories'),
+    },
+    {
+      id: GLOBAL,
+      name: __('All stories', 'web-stories'),
+    },
+  ];
+
+  const savedColors = showLocalColors ? localColors : globalColors;
   return (
     <ActionsWrapper>
-      <AddColorPreset onClick={() => handleAddColorPreset(color)}>
-        {__('+ Add to Color Preset', 'web-stories')}
-      </AddColorPreset>
+      <HeaderRow>
+        <DropDownWrapper>
+          <AdvancedDropDown
+            options={options}
+            isInline={true}
+            hasSearch={false}
+            onChange={({ id }) => setShowLocalColors(id === LOCAL)}
+            selectedId={showLocalColors ? LOCAL : GLOBAL}
+            aria-label={__('Select color type', 'web-stories')}
+          />
+        </DropDownWrapper>
+        <ButtonWrapper>
+          <AddColorPreset
+            aria-label={__('Add color', 'web-stories')}
+            onClick={() => handleAddColorPreset(color)}
+          >
+            <Add />
+          </AddColorPreset>
+        </ButtonWrapper>
+      </HeaderRow>
+      <ColorsWrapper hasColors={savedColors.length > 0}>
+        {savedColors.length > 0 && (
+          <ColorGroup
+            isEditMode={false}
+            isLocal={showLocalColors}
+            colors={savedColors}
+            handleClick={applyStyle}
+            displayAdd={false}
+            colorGap={COLOR_GAP}
+          />
+        )}
+        {!savedColors.length && (
+          <CtaWrapper>
+            <TranslateWithMarkup
+              mapping={{
+                i: <Add width={12} height={12} />,
+                arrow: <Curve width={32} />,
+              }}
+            >
+              {__(
+                'Click <i></i> to save a color <arrow></arrow>',
```

Comment:
Can this arrow be outside the translatable text? Doesn't seem like it belongs inside.

Also, is the arrow position still correct if the translated text is shorter or longer? I doubt it... In that case we should rather remove the arrow entirely IMO

Reply:
Hm, good point. Perhaps the arrow could always be statically in place and the text could use the room only from the left side of the area, not influencing the arrow at all, even if the text goes on two lines for example. But not sure if this would look ok indeed.
Will remove it for now and check with Sam about this.